aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -95,22 +95,70 @@ std::tuple<Tensor, Tensor> slogdet(const Tensor& self) {
   return std::make_tuple(det_sign, abslogdet_val);
 }
 
-Tensor pinverse(const Tensor& self, double rcond) {
-  TORCH_CHECK((at::isFloatingType(self.scalar_type()) || at::isComplexType(self.scalar_type())) && self.dim() >= 2,
-              "pinverse(", self.scalar_type(), "{", self.sizes(), "}): expected a tensor with 2 or more dimensions "
+Tensor linalg_pinv(const Tensor& input, const Tensor& rcond, bool hermitian) {
+  TORCH_CHECK((at::isFloatingType(input.scalar_type()) || at::isComplexType(input.scalar_type())) && input.dim() >= 2,
+              "linalg_pinv(", input.scalar_type(), "{", input.sizes(), "}): expected a tensor with 2 or more dimensions "
               "of floating types");
-  if (self.numel() == 0) {
+  if (input.numel() == 0) {
     // Match NumPy
-    auto self_sizes = self.sizes().vec();
-    std::swap(self_sizes[self.dim() - 1], self_sizes[self.dim() - 2]);
-    return at::empty(self_sizes, self.options());
+    auto input_sizes = input.sizes().vec();
```

Comment:
Add a comment here elaborating on how this matches NumPy

Reply:
Since later in the code, `amax` and `narrow` are used which doesn't work for zero numel tensors. This piece of code was copied from the previous implementation and probably the comment was about NumPy having the same early return for this case. I'll change the comment.
```cpp
    // The implementation below uses operations that do not work for zero numel tensors
    // therefore we need this early return for 'input.numel() == 0' case
```